builder/azure/arm/step_delete_additional_disks.go

```diff
@@ -75,15 +76,16 @@ func (s *StepDeleteAdditionalDisk) Run(ctx context.Context, state multistep.Stat
 		return multistep.ActionContinue
 	}
 
-	if isManagedDisk && !isExistingResourceGroup {
+	// TODO: Not sure about this check
+	if (isManagedDisk || isSIGImage) && !isExistingResourceGroup {
```

Comment:
I think actually you can leave this check as it is, I think we only want to skip deleting additional disks when a Managed Image is being used, since your changes allow us to skip the creation of that image I think this check can stay the same

Reply:
Thinking about this again in the new year, I think this code has always been wrong, look at issue https://github.com/hashicorp/packer-plugin-azure/issues/215 to see why, I don't think this check makes sense personally but I don't think you need to worry about resolving it in your PR, you can leave it like it is in this PR, and I will follow up and see if we can resolve that issue in a separate PR